src/realm/group.cpp

```diff
@@ -511,10 +511,10 @@ void Transaction::upgrade_file_format(int target_file_format_version)
     }
 
     // If we come from a file format version lower than 10, all objects with primary keys
-    // will be upgraded correctly by the above process. In file format 12 we don't have
+    // will be upgraded correctly by the above process. In file format 20 we don't have
     // search index on primary key columns. We need to rebuild the tables to ensure that
     // the ObjKeys matches the primary key value.
-    if (current_file_format_version < 12 && target_file_format_version >= 12) {
+    if (current_file_format_version > 9 && current_file_format_version < 20 && target_file_format_version >= 20) {
```

Comment:
Should the second term be `current_file_format_version <= 12`?

Reply:
No. The strategy is to reserve the range 12..19 to core-6 based file formats.